File: Swift-C++-IR-Comparison/test/functions/Swift/StoreFunctionInVariable.swift
@main
struct MyMain {

    static func main() {
        var mathFunction: (Int, Int) -> Int = addTwoInts
        var result = mathFunction(42, 44)
    }

}

func addTwoInts(_ a: Int, _ b: Int) -> Int {
    return a + b
}

File: Swift-C++-Testsuite/test/functions/C++/StoreFunctionInVariable.cpp

int addTwoInts(int a, int b) {
	return a + b;
}

int main() {
	int (*mathFunction)(int, int); // Declaration of a function pointer
	mathFunction = &addTwoInts;
	int result = mathFunction(42, 44);
	
	return 0;
}

